ui/src/importer/importerTaskDetails/ImportedDocuments.js

```diff
@@ -53,23 +92,132 @@ export class ImportedDocuments extends React.Component {
     if (!importCompleted) {
       const nextEntry = _get(data, 'loaded_entries', 0);
       const response = await importerApi.check(taskId, nextEntry);
+      const responseData = response.data;
+
+      if (responseData) {
```

Comment:
I removed this part before, mainly because it was cause huge problems with pagination, could you test if it works properly ?
Also maybe it is an optimisation for the next PR, but by setting
```
responseData.records = _get(responseData, 'records', []
```
you are just overwriting all the data each time it is refreshed, instead of adding small chunks. The initial problem is that the backend endpoint is returning all the records all the time, instead of respecting the `offset` parameter. If you would have to handle only few records at each refresh, it would be even more justified, to keep each filter under separate state as you do above
ping @ntarocco